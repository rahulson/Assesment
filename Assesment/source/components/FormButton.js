import React from 'react'
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native'
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { ColorConstants } from '../constants/ColorConstants'
import FontSize from '../constants/FontSize'
import PropTypes from 'prop-types';

const FormButton = (props) => {
    const { containerStyle, titleStyle, title = '', onPress } = props
    return (
        <TouchableOpacity style={[styles.container, containerStyle]} onPress={() => onPress()}>
            <Text style={[styles.title, titleStyle]}>{title}</Text>
        </TouchableOpacity>
    )
}

const styles = StyleSheet.create({
    container: {
        width: '100%',
        justifyContent: 'center',
        alignItems: 'center',
        height: hp(6),
        backgroundColor: ColorConstants.PAINT_CHIP,
        borderRadius: hp(1.0)
    },
    title: {
        color: ColorConstants.WHITE,
        fontSize: FontSize.tow,
        fontWeight: '600'
    }
})

FormButton.propTypes = {
    containerStyle: PropTypes.object,
    titleStyle: PropTypes.object,
    title: PropTypes.string,
    onPress: PropTypes.func
};

FormButton.defaultProps = {
    containerStyle: {},
    titleStyle: {},
    title: '',
    onPress: () => { }
};

export default FormButton