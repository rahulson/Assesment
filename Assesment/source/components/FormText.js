import React from 'react'
import { Text, StyleSheet } from 'react-native'
import PropTypes from 'prop-types';

const FormText = (props) => {

    const { textStyle, children, ...restProps } = props

    return (
        <Text style={textStyle} {...restProps}>
            {children}
        </Text>    
    )
}

FormText.propTypes = {
    textStyle: PropTypes.object,
    text: PropTypes.string,
    onPress: PropTypes.func
};

FormText.defaultProps = {
    textStyle: {},
    text: ''
};

export default FormText