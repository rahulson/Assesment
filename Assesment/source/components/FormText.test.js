import React from 'react';
import FormButton from './FormButton'
import { render, fireEvent } from '@testing-library/react-native'
import FormText from './FormText';


test('it should have text as children', () => {
    const text = 'Uni Test';
    const { getByText, toJSON } = render(<FormText>{text}</FormText>);
    const title = getByText(text);
    expect(title).toBeDefined();

    // should match the snapshot
    expect(toJSON()).toMatchSnapshot();
});