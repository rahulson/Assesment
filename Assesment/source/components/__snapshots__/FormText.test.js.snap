// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should have text as children 1`] = `
<Text
  style={Object {}}
  text=""
>
  Uni Test
</Text>
`;
